source/CMakeLists.txt

```diff
@@ -92,10 +94,18 @@ else()
 endif()
 
 if(USE_OPENCV)
-    find_package(OpenCV 4.1.0 COMPONENTS core highgui calib3d aruco)
-    if(NOT OpenCV_FOUND)
-        message(FATAL_ERROR "OpenCV not found. Please point OpenCV_DIR to the directory of your OpenCV installation (containing the file OpenCVConfig.cmake), or disable the OpenCV samples  with -DUSE_OPENCV=OFF.")
-    endif()
+	if(USE_ARUCO)
+		find_package(OpenCV 4.1.0 COMPONENTS core highgui calib3d aruco)
+		if(NOT OpenCV_FOUND)
+			message(FATAL_ERROR "OpenCV not found. Please point OpenCV_DIR to the directory of your OpenCV installation (containing the file OpenCVConfig.cmake), or disable the ArUco samples  with -DUSE_ARUCO=OFF.")
```

Comment:
This might fail because OpenCV was not found at all, in which case it won't help to do `-DUSE_ARUCO=OFF`. But I can't really think of a better way to do it, so I think we can keep doing it like this. And then they'll get the other error message about `USE_OPENCV` when trying again which will fix it.

Reply:
How about I change `, or disable the ArUco samples  with -DUSE_ARUCO=OFF` to `.If you don't have OpenCV with extra modules (OpenCV_Contrib), disable the ArUco samples  with -DUSE_ARUCO=OFF`

Reply:
The user could still get that error message in the case where they don't have OpenCV at all. And then turning off aruco won't help. My comment was just to acknowledge that I saw a potential small issue with this, but that I don't think we need to care about.

That being said, I think your suggested update to the error message is great!

Reply:
done